src/components/PersonItem/PersonItem.tsx

```diff
@@ -0,0 +1,49 @@
+import { useParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { PeopleContex } from '../../peopleContext';
+
+type Props = {
+  person: Person;
+};
+
+const getPersonByName = (
+  name: string | null,
+  people: Person[],
+): Person | undefined => {
+  return people.find(person => person.name === name);
+};
+
+export const PersonItem: React.FC<Props> = ({ person }) => {
+  const { slug: userId, sex, born, died, fatherName, motherName } = person;
+  const { slug: selectedUserId } = useParams();
+
+  const people = useContext(PeopleContex);
+  const mother = getPersonByName(motherName, people);
+  const father = getPersonByName(fatherName, people);
```

Comment:
You are using `getPersonByName` to find the `mother` and `father` of a person. It would be more efficient to use a map data structure to store people by name. This would allow you to look up a person by name in constant time instead of having to iterate over the entire people array for each lookup.